frame/bounties/src/lib.rs

```diff
@@ -0,0 +1,754 @@
+
+	/// Percentage of the curator fee that will be reserved upfront as deposit for bounty curator.
+	type BountyCuratorDeposit: Get<Permill>;
+
+	/// Minimum value for a bounty.
+	type BountyValueMinimum: Get<BalanceOf<Self>>;
+
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+
+	/// Weight information for extrinsics in this pallet.
+	type WeightInfo: WeightInfo;
+}
+
+/// An index of a bounty. Just a `u32`.
+pub type BountyIndex = u32;
+
+/// A bounty proposal.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub struct Bounty<AccountId, Balance, BlockNumber> {
+	/// The account proposing it.
+	proposer: AccountId,
+	/// The (total) amount that should be paid if the bounty is rewarded.
+	value: Balance,
+	/// The curator fee. Included in value.
+	fee: Balance,
+	/// The deposit of curator.
+	curator_deposit: Balance,
+	/// The amount held on deposit (reserved) for making this proposal.
+	bond: Balance,
+	/// The status of this bounty.
+	status: BountyStatus<AccountId, BlockNumber>,
+}
+
+/// The status of a bounty proposal.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub enum BountyStatus<AccountId, BlockNumber> {
+	/// The bounty is proposed and waiting for approval.
+	Proposed,
+	/// The bounty is approved and waiting to become active at next spend period.
+	Approved,
+	/// The bounty is funded and waiting for curator assignment.
+	Funded,
+	/// A curator has been proposed by the `ApproveOrigin`. Waiting for acceptance from the curator.
+	CuratorProposed {
+		/// The assigned curator of this bounty.
+		curator: AccountId,
+	},
+	/// The bounty is active and waiting to be awarded.
+	Active {
+		/// The curator of this bounty.
+		curator: AccountId,
+		/// An update from the curator is due by this block, else they are considered inactive.
+		update_due: BlockNumber,
+	},
+	/// The bounty is awarded and waiting to released after a delay.
+	PendingPayout {
+		/// The curator of this bounty.
+		curator: AccountId,
+		/// The beneficiary of the bounty.
+		beneficiary: AccountId,
+		/// When the bounty can be claimed.
+		unlock_at: BlockNumber,
+	},
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Treasury {
+
+		/// Number of bounty proposals that have been made.
+		pub BountyCount get(fn bounty_count): BountyIndex;
+
+		/// Bounties that have been made.
+		pub Bounties get(fn bounties):
+		map hasher(twox_64_concat) BountyIndex
+		=> Option<Bounty<T::AccountId, BalanceOf<T>, T::BlockNumber>>;
+
+		/// The description of each bounty.
+		pub BountyDescriptions get(fn bounty_descriptions): map hasher(twox_64_concat) BountyIndex => Option<Vec<u8>>;
+
+		/// Bounty indices that have been approved but not yet funded.
+		pub BountyApprovals get(fn bounty_approvals): Vec<BountyIndex>;
+	}
+}
+
+decl_event!(
+	pub enum Event<T>
+	where
+		Balance = BalanceOf<T>,
+		<T as frame_system::Trait>::AccountId,
+	{
+		/// New bounty proposal. [index]
```

Comment:
Needs escape for all the square brackets

```suggestion
		/// New bounty proposal. \[index\]
```

Reply:
Thanks for comment @xlc, have taken the suggested one.